### PR TITLE
JDK-8288094: cleanup old _MSC_VER handling

### PR DESCRIPTION
**Pull request by MBaesken**
We still handle at a number of places ancient historic _MSC_VER versions of Visual Studio releases e.g. pre VS2013 (VS2013 has _MSC_VER 1800).
This should be cleaned up, as long as it is not 3rd party code that we don't want to adjust.

Currently still supported ("valid") VS version are 2017+, see https://github.com/openjdk/jdk/blob/master/make/autoconf/toolchain_microsoft.m4 .
VALID_VS_VERSIONS="2019 2017 2022"
Even with adjusted toolchain m4 files, something older than VS2013 (also probably older than VS2015) would not be able to build jdk anymore.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288094](https://bugs.openjdk.org/browse/JDK-8288094): cleanup old _MSC_VER handling

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to [e97c8329](https://git.openjdk.org/jdk/pull/9105/files/e97c83291a13151f472ae0e6e20cc6689ded3d17)
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**) ⚠️ Review applies to [ee8dc996](https://git.openjdk.org/jdk/pull/9105/files/ee8dc996cd8b769a5a30ab58f60ac9d43391542e)
 * [Andrey Turbanov](https://openjdk.java.net/census#aturbanov) (@turbanoff - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9105/head:pull/9105` \
`$ git checkout pull/9105`

Update a local copy of the PR: \
`$ git checkout pull/9105` \
`$ git pull https://git.openjdk.org/jdk pull/9105/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9105`

View PR using the GUI difftool: \
`$ git pr show -t 9105`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9105.diff">https://git.openjdk.org/jdk/pull/9105.diff</a>

</details>

### CODE REVIEW

**Review by TheRealMDoerr — APPROVED** [review]
LGTM and sounds feasible.

**Comment by MBaesken** [comment]
> LGTM and sounds feasible.

Hi Martin, thanks for the review.

**Review by RealCLanger — APPROVED** [review]
Looks good to me.

File: src/java.desktop/windows/native/libawt/windows/ThemeReader.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 #include <uxtheme.h>
 
-#if defined(_MSC_VER) && _MSC_VER >= 1800
+#if defined(_MSC_VER)
 #  define ROUND_TO_INT(num)    ((int) round(num))
 #else
 #  define ROUND_TO_INT(num)    ((int) floor((num) + 0.5))
```
Comment (prrace) [inline src/java.desktop/windows/native/libawt/windows/ThemeReader.cpp:38]: If you look at when and why this was introduced (*), the "else" was not to support some other compiler - it was to support the older MS compiler. So if you don't want that, then the whole thing reduces to
#define ROUND_TO_INT(num) ((int) round(num))

.. you could even go further and eliminate the macro altogether if it makes sense - you'd have to look at the usages.

Same logic applies to the other files.


(*) https://mail.openjdk.java.net/pipermail/awt-dev/2016-March/010889.html

Comment (MBaesken) [reply src/java.desktop/windows/native/libawt/windows/ThemeReader.cpp]: Hi Phil, I simplified the code more and now use round directly.

**Review by TheRealMDoerr — APPROVED** [review]
Nice!

File: src/hotspot/share/adlc/main.cpp [hunk 2]
```diff
@@ -489,7 +489,7 @@ int get_legal_text(FileBuff &fbuf, char **legal_text)
 }
 
 // VS2005 has its own definition, identical to this one.
```
[the remark this answers is not in the document]
Comment (MBaesken) [reply src/hotspot/share/adlc/main.cpp]: Thanks for the hint, I removed the VS2005 related comment.

**Comment by MBaesken** [comment]
Hi Phil and Andrey are you fine with the current version of the PR ?

**Review by turbanoff — APPROVED** [review]

**Pull request closed without merging**